tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,323 @@
+    expected = pd.DataFrame([2, 3], columns=["x"])
+    assert_frame_equal(_check_instance(data)[0][0], expected)
+
+
+def test_series_not_multi_index_with_name():
+    """Test for single index series with name"""
+    data = {"x": pd.Series([2, 3], name="y")}
+    expected = pd.DataFrame([2, 3], columns=["x_y"])
+    assert_frame_equal(_check_instance(data)[0][0], expected)
+
+
+def test_series_multi_index():
+    """Test that multiIndexed series trigger error"""
+    data = {
+        "x": pd.Series(
+            [2, 3], index=pd.MultiIndex.from_arrays([[1, 2], [3, 4]])
+        )
+    }
+    with pytest.raises(TypeError):
+        expand_grid(others=data)
+
+
+def test_dataframe_empty():
+    """Trigger error for empty dataframes"""
+    data = {"x": pd.DataFrame([])}
+    with pytest.raises(ValueError):
+        expand_grid(others=data)
+
+
+def test_dataframe_single_index():
+    """Test for single indexed dataframes"""
+    data = {"x": pd.DataFrame([[2, 3], [6, 7]])}
+    expected = pd.DataFrame([[2, 3], [6, 7]]).add_prefix("x_")
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_dataframe_multi_index_index():
+    """Trigger error if dataframe has a MultiIndex index"""
+    data = {
+        "x": pd.DataFrame(
+            [[2, 3], [6, 7]],
+            index=pd.MultiIndex.from_arrays([["a", "b"], ["y", "z"]]),
+        )
+    }
+
+    with pytest.raises(TypeError):
+        expand_grid(others=data)
+
+
+def test_dataframe_multi_index_column():
+    """Trigger error if dataframe has a MultiIndex column"""
+    data = {
+        "x": pd.DataFrame(
+            [[2, 3], [6, 7]],
+            columns=pd.MultiIndex.from_arrays([["m", "n"], ["p", "q"]]),
+        )
+    }
+
+    with pytest.raises(TypeError):
+        expand_grid(others=data)
+
+
+def test_dataframe_multi_index_index_and_column():
+    """Trigger error if dataframe has a MultiIndex column or index"""
+    data = {
+        "x": pd.DataFrame(
+            [[2, 3], [6, 7]],
+            index=pd.MultiIndex.from_arrays([["a", "b"], ["y", "z"]]),
+            columns=pd.MultiIndex.from_arrays([["m", "n"], ["p", "q"]]),
+        )
+    }
+    with pytest.raises(TypeError):
+        expand_grid(others=data)
+
+
+def test_list_empty():
+    """Raise error if any list in dictionary's values is empty"""
+    data = {"x": [], "y": [2, 3]}
+    with pytest.raises(ValueError):
+        expand_grid(others=data)
+
+
+def test_lists():
+    """
+    Test expected output
+    from one level nested lists
+    in a dictionary's values
+    """
+    data = {"x": [[2, 3], [4, 3]]}
+    expected = pd.DataFrame([[2, 3], [4, 3]]).add_prefix("x_")
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_lists_all_scalar():
+    """
+    Test that all values in a list
+    in dictionary's values are scalar
+    """
+    data = {"x": [2, 3, 4, 5, "ragnar"]}
+    expected = {"x": [2, 3, 4, 5, "ragnar"]}
+    _check_instance(data)[-1] == expected
```

Comment:
The assertion is necessary in this case.

```suggestion
    assert _check_instance(data)[-1] == expected
```